tests/unit/amo/components/TestSearchResult.js

```diff
@@ -43,14 +43,15 @@ describe(__filename, () => {
     addon = baseAddon,
     history = createFakeHistory(),
     lang = 'en-GB',
+    noAddon = false,
     store = dispatchClientMetadata({
       clientApp: CLIENT_APP_FIREFOX,
     }).store,
     ...props
   } = {}) {
     return shallowUntilTarget(
       <SearchResult
-        addon={addon}
+        addon={noAddon ? undefined : addon}
```

Comment:
Not sure I really understand this change - why is a specific `noAddon` param better than specifying `addon: null`?

Reply:
Sorry, I should have added a comment explaining the changes to the tests. While we have partial Flow coverage for code in addons-frontend, we don't have any Flow coverage for the tests, which resulted in some invalid datatypes being used in this test. 

`addon` can not legally accept a `null`, so we really shouldn't be passing a `null` in any tests. This was done to allow the tests to emulate a `SearchResult` not having an `addon`, but it's not really accurate. A `SearchResult` without an `addon` would have an `addon` of `undefined`, not an `addon` of `null`, so I wanted to fix that in these tests. 

A simple fix would have been to pass `addon: undefined` into the `render` function, but the `render` function is already set up to use an actual `addon` if none is passed, so that tests which want an `addon` to exist don't have to pass one. The problem is that this logic replaces `undefined` with an `addon`, even if you explicitly pass `addon: undefined` into `render`. The easiest thing to do, rather than complicate the `render` function or rewrite other tests, was to add this `noAddon` parameter to the `render` function which can be used to say "render the component without an `addon`".

Reply:
Ah, makes sense.  You can't pass `addon: undefined`.  
(fwiw, the python solution - which happens with `arg=None` defaults - is to have a special constant you can pass in like `addon=NO_ADDON` to indicate you actually want it to be `None`/`undefined`)

Reply:
I suppose I could do that, but I would prefer to have `addon` only accept valid types, so repurposing it for that would kind of defeat that. In code manager we have Typescript for all of the code, including tests, which means that things like this can't happen in the first place. I don't think we'll ever have the time to add Flow coverage to all of our tests in frontend though.